File: src/main/java/com/ecopay/service/auth/AuthService.java
package com.ecopay.service.auth;

import com.ecopay.dto.request.AuthLoginRequest;
import com.ecopay.dto.request.AuthRegisterRequest;
import com.ecopay.dto.request.RefreshTokenRequest;
import com.ecopay.dto.response.AuthResponse;

public interface AuthService {
    AuthResponse register(AuthRegisterRequest request);
    AuthResponse login(AuthLoginRequest request);
    AuthResponse refresh(RefreshTokenRequest request);
    void logout(RefreshTokenRequest request);
}

File: mobile/lib/core/services/auth_service.dart
import '../network/api_client.dart';

class AuthService {
  final _client = ApiClient();

  Future<Map<String, dynamic>> login(String email, String password) async {
    final response = await _client.dio.post('/api/v1/auth/login', data: {
      'email': email,
      'password': password,
    });
    final data = response.data as Map<String, dynamic>;
    await ApiClient.saveTokens(
      data['accessToken'] as String,
      data['refreshToken'] as String,
    );
    return data;
  }

  Future<Map<String, dynamic>> register(String email, String password) async {
    final response = await _client.dio.post('/api/v1/auth/register', data: {
      'email': email,
      'password': password,
    });
    final data = response.data as Map<String, dynamic>;
    await ApiClient.saveTokens(
      data['accessToken'] as String,
      data['refreshToken'] as String,
    );
    return data;
  }

  Future<void> logout() async {
    final refreshToken = await ApiClient.getRefreshToken();
    await ApiClient.clearTokens();
    if (refreshToken != null && refreshToken.isNotEmpty) {
      try {
        await _client.dio.post('/api/v1/auth/logout', data: {
          'refreshToken': refreshToken,
        });
      } catch (_) {
        // Sunucu cevap vermese bile yerel oturum zaten kapandı.
      }
    }
  }
}
